### PR TITLE
Cherry-pick check titandb directory when titan is disabled

### PR DESCRIPTION
**Pull request by tabokie**
Signed-off-by: tabokie <xy.tao@outlook.com>

###  What have you changed?

Check titandb directory is empty (`ls *.blob` count = 0) when titan is disabled. Fatal panic if this check fails, which indicates titan is turned off before all blob files are merged.

###  What is the type of the changes?

- Misc (other changes)

###  How is the PR tested?

- Unit test

### CODE REVIEW

**Comment by Connor1996** [comment]
Please do not merge it until we release v3.0.5

File: components/test_raftstore/src/cluster.rs [hunk 1]
```diff
@@ -127,18 +128,26 @@ impl<T: Simulator> Cluster<T> {
...
+            let kv_path = dir.path().join(DEFAULT_ROCKSDB_SUB_DIR);
             let cache = self.cfg.storage.block_cache.build_shared_cache();
             let kv_db_opt = self.cfg.rocksdb.build_opt();
             let kv_cfs_opt = self.cfg.rocksdb.build_cf_opts(&cache);
             let engine = Arc::new(
                 rocks::util::new_engine_opt(kv_path.to_str().unwrap(), kv_db_opt, kv_cfs_opt)
                     .unwrap(),
             );
-            let raft_path = dir.path().join(Path::new("raft"));
+            let raft_path = dir.path().join("raft");
```
Comment (AndreMouche) [inline components/test_raftstore/src/cluster.rs]: what's the differance?

Comment (tabokie) [reply components/test_raftstore/src/cluster.rs]: no difference, can remove the `Path` dependency now.

**Pull request merged**